src/client/debugger/jupyter/kernelDebugAdapter.ts

```diff
@@ -122,6 +123,22 @@ export class KernelDebugAdapter implements DebugAdapter, IKernelDebugAdapter, ID
                 this.disposables
             )
         );
+
+        this.disposables.push(
+            notebooks.onDidChangeNotebookCells(
+                (e: NotebookCellsChangeEvent) => {
+                    e.changes.forEach((change) => {
+                        change.deletedItems.forEach((cell: NotebookCell) => {
+                            if (cell.document.uri === this.debugCellUri) {
```

Comment:
Rather than storing the `debugCellUri`, you might as well store the reference to the cell object.
Then just compare `if (cell === this.debugCell)`
that will be better (else you end up comparing uris everytime).